From a82e6efb7b9b2ab9a1597b002f375c5ee105e7f5 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 16 Jul 2008 18:19:04 +0000
Subject: Remove the numVideoScreens xprintism.

This was to account for cases where you had video and print screens in
the same server.  Lunacy.  Leave the slot in ScreenInfo, but rename it,
and stop looking at it.
---
diff --git a/dix/dixfonts.c b/dix/dixfonts.c
index 89f1528..1c6d37f 100644
--- a/dix/dixfonts.c
+++ b/dix/dixfonts.c
@@ -1908,21 +1908,12 @@ InitFonts (void)
 {
     patternCache = MakeFontPatternCache();
 
-#ifndef BUILTIN_FONTS
-    if (screenInfo.numScreens > screenInfo.numVideoScreens) {
-	PrinterFontRegisterFpeFunctions();
-	FontFileCheckRegisterFpeFunctions();
-	check_fs_register_fpe_functions();
-    } else 
-#endif
-    {
 #ifdef BUILTIN_FONTS
-        BuiltinRegisterFpeFunctions();
+    BuiltinRegisterFpeFunctions();
 #else
-	FontFileRegisterFpeFunctions();
-	fs_register_fpe_functions();
+    FontFileRegisterFpeFunctions();
+    fs_register_fpe_functions();
 #endif
-    }
 }
 
 int
diff --git a/dix/main.c b/dix/main.c
index 3bc45fb..d8e3bac 100644
--- a/dix/main.c
+++ b/dix/main.c
@@ -309,7 +309,6 @@ int main(int argc, char *argv[], char *envp[])
 	SetInputCheck(&alwaysCheckForInput[0], &alwaysCheckForInput[1]);
 	screenInfo.arraySize = MAXSCREENS;
 	screenInfo.numScreens = 0;
-	screenInfo.numVideoScreens = -1;
 	WindowTable = (WindowPtr *)xalloc(MAXSCREENS * sizeof(WindowPtr));
 	if (!WindowTable)
 	    FatalError("couldn't create root window table");
@@ -339,8 +338,6 @@ int main(int argc, char *argv[], char *envp[])
 
 	if (screenInfo.numScreens < 1)
 	    FatalError("no screens found");
-	if (screenInfo.numVideoScreens < 0)
-	    screenInfo.numVideoScreens = screenInfo.numScreens;
 	InitExtensions(argc, argv);
 	for (i = 0; i < screenInfo.numScreens; i++)
 	{
diff --git a/include/scrnintstr.h b/include/scrnintstr.h
index 6cf9333..5d4ddbb 100644
--- a/include/scrnintstr.h
+++ b/include/scrnintstr.h
@@ -612,7 +612,7 @@ typedef struct _ScreenInfo {
     int		arraySize;
     int		numScreens;
     ScreenPtr	screens[MAXSCREENS];
-    int		numVideoScreens;
+    int		unused;
 } ScreenInfo;
 
 extern ScreenInfo screenInfo;
--
cgit v0.8.2
